### PR TITLE
feat: tinywikitext dataset

### LINKED ISSUE #356: [FEATURE] Add TinyWikiText Dataset
## **✨ What You’ll Do**

Data plays a key role in running algorithms and evaluating models in Pruna. To make this easier, we provide pre-configured datasets for different tasks. However, these datasets can be quite large, and smaller models or quick POCs don’t always need that much data.

In this issue, your task is to add a tiny version of the WikiText dataset (fewer than 1,000 rows) that can be used for lightweight experiments and testing.

## 🤖 Useful Resources
For the details on how to implement this dataset, check out our guide: [Customize Datasets](https://docs.pruna.ai/en/stable/docs_pruna/user_manual/customize_dataset.html)

**Main steps**:
- [ ]  1. Search for or create your dataset. Recommendation: Save your dataset in HF so you can easily load it later.
- [ ] 2. Add the setup method.
- [ ] 3. Register the dataset in the `base_datasets` dictionary. Attention: Ensure the dataset follows the expected format specified in the collate function.
- [ ] 4. Add a test

## ➡ Desired Solution

```py
from pruna import SmashConfig

smash_config = SmashConfig
smash_config.add_data("TinyWikiText")
```

## **✅ Acceptance Criteria**

- **Tests & Docs:** All existing and new unit tests pass, and the documentation is updated

And don’t forget to give us a ⭐️!

## **❓ Questions?**

Feel free to jump into the [#contributing Discord channel](https://discord.com/invite/JFQmtFKCjd) if you hit any roadblocks. Can’t wait to see your contribution! 🚀

---

## Share on X

[Share on X](https://twitter.com/intent/tweet?text=I%20just%20contributed%20to%20%40PrunaAI%20open%20source%20project!%20%F0%9F%9A%80%20Check%20out%20my%20PR%20%F0%9F%91%89%20%5BADD%20LINK%20TO%20PR%5D)

### PR DESCRIPTION
**Pull request by R055A**
## Description

Add feature: TinyWikiText dataset

## Related Issue

Fixes #356 

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Unit testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Additional Notes

---

> [!NOTE]
> Adds `TinyWikiText` dataset with configurable small splits and updates `WikiText` loader types/returns; integrates into registry and tests.
> 
> - **Datasets (text generation)**:
>   - **New**: `setup_wikitext_tiny_dataset(seed=42, num_rows=960)` creates a tiny `WikiText` split (.8/.1/.1) via shuffle/select with size assertions.
>   - **Update**: `setup_wikitext_dataset` now returns a `Tuple[Dataset, Dataset, Dataset]` with explicit train/val/test variables; switches to `datasets.Dataset` typing and import cleanup.
> - **Registry**:
>   - Add `"TinyWikiText"` to `base_datasets` in `src/pruna/data/__init__.py` using `text_generation_collate`.
> - **Tests**:
>   - Add `TinyWikiText` to `tests/data/test_datamodule.py::test_dm_from_string` parametrization with `bert_tokenizer`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7aabc1fa028895ee304a4fc7a3ee11c4a3e10b63. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by johannaSommer** [comment]
@R055A I ran the linting & tests, can you address the failures? Then we can merge :)

**Comment by r055a** [comment]
> @R055A I ran the linting & tests, can you address the failures? Then we can merge :)

Hi @johannaSommer the issue is that `Dataset` is being imported from `torch.utils.data` when it should be imported from hugging face `datasets`.

```
Run mypy --show-traceback src/pruna
src/pruna/data/datasets/text_generation.py:67: error: Argument 1 to "len" has incompatible type "Dataset[Any]"; expected "Sized"  [arg-type]
src/pruna/data/datasets/text_generation.py:68: error: Argument 1 to "len" has incompatible type "Dataset[Any]"; expected "Sized"  [arg-type]
src/pruna/data/datasets/text_generation.py:69: error: Argument 1 to "len" has incompatible type "Dataset[Any]"; expected "Sized"  [arg-type]
src/pruna/data/datasets/text_generation.py:72: error: "Dataset[Any]" has no attribute "shuffle"  [attr-defined]
src/pruna/data/datasets/text_generation.py:73: error: "Dataset[Any]" has no attribute "shuffle"  [attr-defined]
src/pruna/data/datasets/text_generation.py:74: error: "Dataset[Any]" has no attribute "shuffle"  [attr-defined]
Found 6 errors in 1 file (checked 87 source files)
Error: Process completed with exit code 1.
```

**Comment by r055a** [comment]
> > @R055A I ran the linting & tests, can you address the failures? Then we can merge :)
> 
> Hi @johannaSommer the issue is that `Dataset` is being imported from `torch.utils.data` when it should be imported from hugging face `datasets`.
> 
> ```
> Run mypy --show-traceback src/pruna
> src/pruna/data/datasets/text_generation.py:67: error: Argument 1 to "len" has incompatible type "Dataset[Any]"; expected "Sized"  [arg-type]
> src/pruna/data/datasets/text_generation.py:68: error: Argument 1 to "len" has incompatible type "Dataset[Any]"; expected "Sized"  [arg-type]
> src/pruna/data/datasets/text_generation.py:69: error: Argument 1 to "len" has incompatible type "Dataset[Any]"; expected "Sized"  [arg-type]
> src/pruna/data/datasets/text_generation.py:72: error: "Dataset[Any]" has no attribute "shuffle"  [attr-defined]
> src/pruna/data/datasets/text_generation.py:73: error: "Dataset[Any]" has no attribute "shuffle"  [attr-defined]
> src/pruna/data/datasets/text_generation.py:74: error: "Dataset[Any]" has no attribute "shuffle"  [attr-defined]
> Found 6 errors in 1 file (checked 87 source files)
> Error: Process completed with exit code 1.
> ```

This is now fixed.

**Comment by johannaSommer** [comment]
@R055A I reran, some linting is still failing, can you have a look? you can also run this locally very fast by running `ruff check src/`! 😊

**Comment by r055a** [comment]
> @R055A I reran, some linting is still failing, can you have a look? you can also run this locally very fast by running `ruff check src/`! 😊

Hi @johannaSommer sorry about that. It now passes the ruff lint check locally

**Review by sdiazlor — APPROVED** [review]
Thanks @R055A for the contribution. Looks great!
